src/terrain/services/bootstrap.clj

```diff
@@ -0,0 +1,71 @@
+(ns terrain.services.bootstrap
+  (:use
+    [slingshot.slingshot :only [try+]]
+    [terrain.auth.user-attributes :only [current-user]])
+  (:require
+    [clojure.tools.logging :as log]
+    [terrain.clients.apps :as apps-client]
+    [terrain.clients.data-info :as data-info-client]
+    [terrain.services.user-prefs :as prefs]
+    [terrain.util.service :as service]))
+
+(defn- decode-error-response
+  [body]
+  (try+
+    (service/decode-json body)
+    (catch Throwable e
+      body)))
+
+(defn- trap-bootstrap-request
+  [req]
+  (try+
+    (req)
+    (catch #(not (nil? (:status %))) {:keys [status body] :as e}
+      (log/error e)
+      {:status status
+       :error  (decode-error-response body)})
+    (catch Throwable e
```

Comment:
It might be worth putting a `catch Object e` somewhere in here too. slingshot can throw basically anything, more than just Throwables, and it's possible we're using that somewhere.